### PR TITLE
Fix/update gemfile generation and rubocop yml

### PR DESCRIPTION
**Pull request by anamarijabalaban**
## Summary
Fix rubocop issues regarding duplicated groups in Gemfile and rubocop-infinum plugin support.

**Related issue**:   None

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [x ] **Bug fix**: This pull request fixes a bug.
- [ ] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

### Description

It looks like the 'rails new' command adds development and test groups by default, so the process of adding another gems in these groups should be updated. Also, rubocop.yml has been updated because of changes in the rubocop-infinum gem regarding using plugins.

## Checklist

- [x ] I have performed a self-review of my own code.
- [ x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

File: recipes/gemfile.rb [hunk 1]
```diff
@@ -13,25 +13,19 @@
     gem 'strong_migrations'
   HEREDOC
 end
-append_to_file 'Gemfile' do
+append_to_file 'Gemfile', after: /group :development do\n/ do
```
Comment (vr4b4c) [inline recipes/gemfile.rb:16]: Does this work for you? I've tried it on Ruby 3.4.1 with Rails 8.0.2 and it's not working because new app doesn't have a "development-only" bundler group.

File: recipes/gemfile.rb [hunk 2]
```diff
@@ -13,25 +13,19 @@
...
+    gem 'annotate'
+    gem 'better_errors'
+    gem 'binding_of_caller'
+    gem 'bullet'
+    gem 'overcommit', require: false
+    gem 'thor', require: false
   HEREDOC
 end
-append_to_file 'Gemfile' do
+append_to_file 'Gemfile', after: /group :test do\n/ do
```
Comment (vr4b4c) [inline recipes/gemfile.rb:26]: Same problem here, new app doesn't have "test-only" Bundler group.

File: recipes/gemfile.rb [hunk 3]
```diff
@@ -13,27 +13,50 @@
...
 
-    group :development do
-      gem 'annotate'
-      gem 'better_errors'
-      gem 'binding_of_caller'
-      gem 'bullet'
-      gem 'overcommit', require: false
-      gem 'thor', require: false
-    end
+def development_group_gems
```
Comment (vr4b4c) [inline recipes/gemfile.rb:17]: I'd probably pick local variable instead of global method

**Review by vr4b4c — APPROVED** [review]

**Comment by vr4b4c** [comment]
I was considering testing the template against the new app generated using the `--minimal` flag, and I believe it was the wrong choice. I don't think it'll be commonly used in practice. I'll revisit this particular topic in another PR.
